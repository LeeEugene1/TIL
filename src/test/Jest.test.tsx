import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import Jest from "./Jest";
import renderer from "react-test-renderer";

// https://www.youtube.com/watch?v=L1dtkLeIz-M
// https://www.youtube.com/watch?v=ML5egqL3YFE

//npm run test:watch

// test("renders Jest Component and find jest", () => {
//   render(<Jest />);
//   const linkElement = screen.getByText(/jest/i);
//   expect(linkElement).toBeInTheDocument();
// });

afterEach(() => {
  cleanup();
});

describe("renders Jest Component", () => {
  it("sould render completed", () => {
    const tasks = [{ id: 1, title: "유진", completed: true }];
    //방법1 getByTestId
    render(<Jest tasks={tasks} />);
    const elem = screen.getByTestId("todo-1");
    expect(elem).toHaveTextContent("유진");
    // expect(elem).toHaveTextContent("건물주");
    expect(elem).toContainHTML("em");
  });

  it("sould render not completed", () => {
    const tasks = [
      {
        id: 2,
        title: "건물주",
        completed: false,
      },
    ];
    //방법2 container
    const { container } = render(<Jest tasks={tasks} />);
    expect(container).toHaveTextContent("건물주");
    expect(container).not.toContainHTML("em");
  });
});

test("maches snapshot", () => {
  const tasks = [{ id: 1, title: "유진", completed: true }];

  const tree = renderer.create(<Jest tasks={tasks} />).toJSON();
  expect(tree).toMatchSnapshot();
});
