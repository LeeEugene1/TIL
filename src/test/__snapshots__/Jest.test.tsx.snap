// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshot 1`] = `
<ul>
  <li
    data-testid="todo-1"
  >
    <em>
      유진
    </em>
  </li>
</ul>
`;
